crates/aptos-telemetry-service/src/peer_location.rs

```diff
@@ -48,28 +48,38 @@ impl PeerLocationUpdater {
     }
 }
 
+
+
 pub async fn query_peer_locations(
     client: &BigQueryClient,
 ) -> anyhow::Result<HashMap<PeerId, PeerLocation>> {
     let req = QueryRequest::new("
+        WITH latest_validator_pool AS (
         SELECT
-            sq.peer_id,
-            sq.country,
-            sq.region,
-            '1985-04-12T23:20:50.52Z' as geo_updated_at
-        FROM (
-            SELECT
-            tm.peer_id,
-            tm.epoch,
-            ROW_NUMBER() OVER (PARTITION BY tm.peer_id ORDER BY tm.epoch DESC) AS row_number,
-            tm.country,
-            tm.region
-            FROM
-            `node-telemetry.aptos_node_telemetry.custom_events_mainnet_telemetry_rollup_metrics` tm) sq
+            CONCAT('0x', LPAD(LTRIM(peer_id, '0x'), 64, '0')) as peer_id
+        FROM `node-telemetry.aptos_data.validator_perf`
+        WHERE chain_id = 1
+        AND epoch = (SELECT MAX(epoch) FROM `node-telemetry.aptos_data.validator_perf` WHERE chain_id = 1)
+        ), latest_node_ips AS (
+        SELECT DISTINCT
+            chain_id,
+            epoch,
+            element.value as ip,
+            -- event_name,
+            CONCAT('0x', LPAD(LTRIM(peer_id, '0x'), 64, '0')) as peer_id
+        FROM `analytics-test-345723.aptos_node_telemetry.custom_events`,
+        UNNEST(event_params) as element
         WHERE
-            sq.row_number = 1
-        LIMIT
-            1000
+            chain_id = \"1\"
+            AND element.key = 'IP_ADDRESS'
+            -- AND event_name = 'APTOS_NODE_BUILD_INFORMATION'
+            AND epoch = (SELECT MAX(epoch) FROM `node-telemetry.aptos_data.validator_perf` WHERE chain_id = 1)
```

Comment:
do we have data for other chain IDs to support here? In general it would be good to make this query generic over a chain_id so we can add support for other chains when needed.

Reply:
Updated the query to include other chain_ids

Reply:
I think we only look up geo for mainnet ips right now